src/Hazelcast.Net.Examples/Sql/LINQExample.cs

```diff
@@ -0,0 +1,128 @@
+            var mayorDC = new Mayor() {MayorName = "Marvin Gaye", Age = 45};
+            await map.PutAsync(cityDC, mayorDC);
+
+            var cityCA = new City() {Code = 3, CityName = "California", IsCapital = false};
+            var mayorCA = new Mayor() {MayorName = "Dave Brubeck", Age = 92};
+            await map.PutAsync(cityCA, mayorCA);
+
+            // Create mapping to be able to run SQL queries over the map.
+            // More details: https://docs.hazelcast.com/hazelcast/latest/sql/mapping-to-maps
+            // Note: Column names are case sensitive. Hazelcast Linq provider passes property names as it is. 
+            await client.Sql.ExecuteCommandAsync(
+                "CREATE OR REPLACE MAPPING cityMayor (" +
+                "Code INT EXTERNAL NAME \"__key.Code\"," +
+                "CityName VARCHAR EXTERNAL NAME \"__key.CityName\"," +
+                "IsCapital BOOLEAN EXTERNAL NAME \"__key.IsCapital\"," +
+                "MayorName VARCHAR," +
+                "Age INT) " +
+                "TYPE IMap OPTIONS (" +
+                "'keyFormat' = 'compact'," +
+                "'keyCompactTypeName' = 'city'," +
+                "'valueFormat' = 'compact'," +
+                "'valueCompactTypeName' = 'mayor')"
+            );
+
+            // Now, we can run SQL queries.
+            var capitalsAndMayors = map.AsAsyncQueryable()
+                .Where(p => p.Key.IsCapital)
+                .Select(p => new {City = p.Key.CityName, Mayor = p.Value.MayorName});
+
+            await foreach (var entry in capitalsAndMayors)
+                logger.LogInformation($"City: {entry.City}, Mayor: {entry.Mayor}");
+
+            // SQL Statement: "SELECT m0.CityName, m0.MayorName FROM cityMayor m0 WHERE m0.IsCapital != FALSE"
+            // Output:
+            // City: Ankara, Mayor: Zeki Müren
+            // City: Washington, DC, Mayor: Marvin Gaye
+
+
+            var youngMayors = map.AsAsyncQueryable()
+                .Where(p => p.Value.Age < 50);
+
+            await foreach (var entry in youngMayors)
+                logger.LogInformation($"City: {entry.Key.CityName}, Mayor: {entry.Value.MayorName}, Age: {entry.Value.Age}");
+
+            // SQL Statement: "SELECT m0.CityName, m0.Code, m0.IsCapital, m0.MayorName, m0.Age FROM cityMayor m0 WHERE (m0.Age < ?)"
+            // Output:
+            // City: Ankara, Mayor: Zeki Müren, Age: 35
+            // City: Washington, DC, Mayor: Marvin Gaye, Age: 45
+
+            var zMuren = map.AsAsyncQueryable()
+                .Where(p => p.Value.MayorName == "Zeki Müren")
+                .Select(p => p.Value.MayorName);
+
+            await foreach (var entry in zMuren)
+                logger.LogInformation($"Mayor: {entry}");
+
+            // SQL Statement: "SELECT m0.MayorName FROM cityMayor m0 WHERE (m0.MayorName = ?)"
+            // Output:
+            // Mayor: Zeki Müren
+        }
+
+        public class City
```

Comment:
Just for the sake of the demo do you think we could make one of these a `record` ie:
```
public record City (string CityName, int Code, bool IsCapital);
```
It should work with Compact Serialization and it's a nice demo.

Reply:
Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/759/commits/eedb97b777a60525334a90a76da72e794066f235.